### PR TITLE
GET-908 Allow passing in ipSecurityRestrictions per webapp

### PR DESCRIPTION
**Pull request by mafonso**
### Context
https://dfedigital.atlassian.net/browse/GET-908

### Changes proposed in this pull request
This PR allows passing an ipSecurityRestrictions object for each of the WebApps (app and admin) 

### Guidance to review

Uses https://github.com/DFE-Digital/get-help-to-retrain-devops/commit/6cc89a562e4973a4b73b12ceb930071892c1606e

### CODE REVIEW

File: azure/config/dev1.parameters.json [hunk 1]
```diff
@@ -43,6 +43,30 @@
...
+    "appIpSecurityRestrictions": {
+      "value": [
+        {
+          "ipAddress": "147.243.0.0/16",
+          "name": "FrontDoor ingress IPv4"
+        },
+        {
+          "ipAddress": "2a01:111:2050::/44",
+          "name": "FrontDoor ingress IPv6"
+        },
```
Comment (mafonso) [inline azure/config/dev1.parameters.json]: From https://docs.microsoft.com/en-us/azure/frontdoor/front-door-faq#how-do-i-lock-down-the-access-to-my-backend-to-only-azure-front-door

File: azure/config/qa.parameters.json [hunk 2]
```diff
@@ -43,6 +43,29 @@
         "secretName": "qa-restrictedDatabasePassword"
       }
     },
+    "appIpSecurityRestrictions": {
+      "value": [
+        {
+          "ipAddress": "147.243.0.0/16",
+          "name": "FrontDoor ingress IPv4"
+        },
```
Comment (mafonso) [inline azure/config/qa.parameters.json]: From: https://docs.microsoft.com/en-us/azure/frontdoor/front-door-faq#how-do-i-lock-down-the-access-to-my-backend-to-only-azure-front-door

File: azure/config/qa.parameters.json [hunk 3]
```diff
@@ -43,6 +43,29 @@
...
+        {
+          "ipAddress": "2a01:111:2050::/44",
+          "name": "FrontDoor ingress IPv6"        },
+        {
```
Comment (mafonso) [inline azure/config/qa.parameters.json]: From https://docs.microsoft.com/en-us/azure/frontdoor/front-door-faq#how-do-i-lock-down-the-access-to-my-backend-to-only-azure-front-door

File: azure/template.json [hunk 4]
```diff
@@ -94,6 +94,20 @@
         "description": "List of IP Addresses to whitelist on the database firewall"
       }
     },
+    "appIpSecurityRestrictions": {
+      "type": "array",
+      "defaultValue": [],
+      "metadata": {
+        "description": "List of IP Address to whitlist on the Application instance Azure Webapp service "
```
Comment (tavy88) [inline azure/template.json]: Typo here: `whitlist`

File: azure/template.json [hunk 5]
```diff
@@ -94,6 +94,20 @@
...
+      }
+    },
+    "adminIpSecurityRestrictions": {
+      "type": "array",
+      "defaultValue": [],
+      "metadata": {
+        "description": "List of IP Address to whitlist on the Admin instance Azure Webapp service "
```
Comment (tavy88) [inline azure/template.json]: Typo here: `whitlist`

File: azure/config/dev1.parameters.json [hunk 6]
```diff
@@ -43,6 +43,22 @@
         "secretName": "dev-restrictedDatabasePassword"
       }
     },
+    "appIpSecurityRestrictions": {
+      "value": [
+        {
+          "ipAddress": "217.38.175.68/32",
```
Comment (cwrw) [inline azure/config/dev1.parameters.json]: Don't office IPs change, are we not risking to have to change the code when that happens?

File: azure/config/qa.parameters.json [hunk 7]
```diff
@@ -43,6 +43,29 @@
...
+          "ipAddress": "217.38.175.68/32",
+          "name": "Office"
+        }
+      ]
+    },
+    "adminIpSecurityRestrictions": {
+      "value": [
+        {
+          "ipAddress": "217.38.175.68/32",
```
Comment (tavy88) [inline azure/config/qa.parameters.json]: Discussed in person about not doing this code wise and doing it from UI but I think you're right we should review changes like this anyway.

File: azure/config/dev1.parameters.json (hunk 6 above)
Comment (mafonso) [reply azure/config/dev1.parameters.json]: Yes that is correct. Ideally I would even add the office address here, but I can see how it is convenient.  But if it changes or you need to add more ranges to the admin area (likely) they should be added as needed.

Comment (tavy88) [reply azure/config/dev1.parameters.json]: Ok

Comment (cwrw) [reply azure/config/dev1.parameters.json]: discussed in person: agreed to add range instead, and confirm that DFE kit can access as well

**Comment by tavy88** [comment]
Discussed in person and happy with these rules for more flexible access.

**Pull request merged**